processor/src/lib.rs

```diff
@@ -255,14 +255,29 @@ where
             MastNode::Loop(node) => self.execute_loop_node(node, program),
             MastNode::Call(node) => self.execute_call_node(node, program),
             MastNode::Dyn => self.execute_dyn_node(program),
+            MastNode::External(external_node) => {
+                let mast_forest =
+                    self.host.borrow().get_mast_forest(&external_node.digest()).ok_or_else(
+                        || ExecutionError::MastForestNotFound {
+                            root_digest: external_node.digest(),
+                        },
+                    )?;
+                let root_id = mast_forest.find_procedure_root(external_node.digest()).ok_or(
```

Comment:
I would think that an `External` node could be _any_ MAST node, not necessarily a procedure root. In particular, this would make it impossible to hide portions of a program - you would have to factor out all of the parts you want to hide into procedures, which sort of defeats the point. That was in fact what `Proxy` was used for previously, and I've somewhat lost the thread on what our plans are for the use cases it supported.

Maybe this is a fine restriction to have for now, but we should probably make a note here that the implicit assumption here that `External` nodes are always procedures is an artifact of other implementation choices, and not of the representation itself. In particular, the notion that `exec` is restricted to invoking procedures is a fiction - if you specify the MAST root of a non-procedure node as the callee of `exec`, that is totally valid, because the semantics are that the referenced MAST is treated as inlined at the callsite. This is in fact the only current means by which someone can hide portions of their program in Miden Assembly today.

Anyway, I'm bringing this up, because I know we made various claims about what the assembler does and does not produce when we were discussing `Proxy`, but I think what got lost in that discussion was what the prior representation made possible, that I'm not sure is being preserved by the changes made as part of the MAST refactoring. That may very well be deliberate, but it is not documented AFAICT.